Data validation + automatic test suites

clbarnes opened the pull request:
There are some test suites which are wrong, and some validation I have yet to implement (plate schema), but this gets further along the way and shows validatrix in action.

clbarnes commented:
I've bumped the MSRV from 1.75 to 1.79 for the sake of `monostate`.

LDeakin commented:
Cool stuff. I've not yet fully given this a proper look (on holidays), but will soon. What do you think is the path forward for this? Just skip the few failing test cases for now?

Since `validatrix` is quite new and is going to be a public dependency of this crate, could you aim to add MSRV/semver checks to CI and keep a changelog?

clbarnes commented:
There are still some holes to fill in validators I haven't got to yet - so far it's been a free time project but will be easier to devote job time to in a few weeks. Then yeah, I think it's easy enough to exclude the test cases we suspect are incorrect.

I'll look into adding those checks to validatrix too! https://github.com/clbarnes/validatrix/issues/1

clbarnes commented:
I've added support for the omero metadata and implemented the plate validation, which means that all conformance tests now pass, except the ones which are wrong, listed here https://github.com/ome/ngff/issues/325 .

The semver check fails for 2 reasons. Firstly, it doesn't seem to run the build.rs which generates the conformance test files, so the tests throw a compilation error. Secondly, it should fail, because I've changed some field names for consistency (they might change anyway if the popularity of https://github.com/ome/ngff/issues/347 is to be believed).

clbarnes commented:
Thanks for the finishing touches! Good find on only running build.rs in test mode, I looked for that switch but hadn't found it.
